testkit/src/checkpoint_db.rs

```diff
@@ -195,33 +218,83 @@ mod tests {
     }
 
     #[test]
-    fn test_checkpoint_db_basics() {
+    fn test_backup_stack() {
         let db = CheckpointDb::new(MemoryDB::new());
+        let handler = db.handler();
+
+        {
```

Comment:
nitpick: you can move these repeating pieces into helper function.

Reply:
I originally decided that this helper function will hurt consistency with code in latter tests (since they have a similar pattern which cannot be replaced by the same helper function). But on second thought, introducing a helper is a good idea. Will do